xdsl/dialects/builtin.py

```diff
@@ -389,6 +389,23 @@ def from_int_and_width(value: int, width: int) -> IntegerAttr[IntegerType]:
     def from_index_int_value(value: int) -> IntegerAttr[IndexType]:
         return IntegerAttr(value, IndexType())
 
+    def verify(self) -> None:
+        if isinstance(self.typ, IntegerType):
+            if self.typ.signedness.data == Signedness.SIGNED:
```

Comment:
match here? Seems like it would be a bit cleaner, with an `assert False` for the default case